testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js

```diff
@@ -180,4 +191,209 @@ QUnit.module('DateRangeBox Initialization', moduleConfig, () => {
+                assert.deepEqual(expectedOptions[key], startDateBox.option(key), `${key} value is correct`);
+            });
+        });
+
+        QUnit.test('EndDateBox has expected settings', function(assert) {
+            this.reinit(initialDateRangeBoxOptions);
+
+            const expectedOptions = {
+                showClearButton: true,
+                showDropDownButton: false,
+                buttons: undefined,
+            };
+            const endDateBox = getEndDateBoxInstance(this.instance);
+
+            Object.entries(initialDateRangeBoxOptions).forEach(([key]) => {
+                assert.deepEqual(expectedOptions[key], endDateBox.option(key), `${key} value is correct`);
+            });
+        });
+    });
+});
+
+QUnit.module('DropDownButton', moduleConfig, () => {
+    QUnit.test('Drop down button should be rendered if showDropDownButton is true', function(assert) {
+        this.reinit({
+            showDropDownButton: true
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 1, 'drop down button was rendered');
+    });
+
+    QUnit.test('Drop down button should not be rendered if showDropDownButton is false', function(assert) {
+        this.reinit({
+            showDropDownButton: false
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 0, 'drop down button was not rendered');
+    });
+
+    QUnit.test('Drop button should not be rendered if buttons is [] and showDropDownButton is true', function(assert) {
+        this.reinit({
+            showDropDownButton: true,
+            buttons: []
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 0, 'drop down button was not rendered');
+    });
+
+    QUnit.test('Drop button should not be rendered if buttons is ["dropDown"] and showDropDownButton is false', function(assert) {
+        this.reinit({
+            showDropDownButton: false,
+            buttons: ['dropDown']
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 0, 'drop down button was not rendered');
+    });
+
+    QUnit.test('Drop button should be rendered if buttons is ["dropDown"]', function(assert) {
+        this.reinit({
+            buttons: ['dropDown']
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 1, 'drop down button was rendered');
+    });
+
+    QUnit.test('Drop button template should be rendered correctly', function(assert) {
+        const buttonTemplate = function(buttonData, contentElement) {
+            assert.strictEqual(isRenderer(contentElement), !!config().useJQuery, 'contentElement is correct');
+
+            return '<div>Template</div>';
+        };
+
+        this.reinit({
+            dropDownButtonTemplate: buttonTemplate
+        });
+
+        assert.strictEqual(getButtons(this.$element).text(), 'Template', 'template was rendered');
+    });
+
+    QUnit.test('Drop button template should be changed correctly in runtime', function(assert) {
```

Comment:
shouldn't we also test runtime change of other props?

Reply:
I added tests for `buttons` & `showDropDownButton` options change in runtime https://github.com/DevExpress/DevExtreme/pull/24306/commits/0ece794aafd40022a99d12436a248615e63456f5